tracer/transport_test.go

```diff
@@ -59,3 +58,57 @@ func TestTracesAgentIntegration(t *testing.T) {
 		assert.Equal(200, response.StatusCode)
 	}
 }
+
+func TestAPIDowngrade(t *testing.T) {
+	assert := assert.New(t)
+	transport := newHTTPTransport("http://localhost:7777/v0.0/traces")
+
+	// if we get a 404 we should downgrade the API
+	traces := getTestTrace(2, 2)
+	response, err := transport.Send(traces)
```

Comment:
Unfortunately we didn't ship the agent open source (but it will be available soon). The CI uses an [internal build][1] and that's why tests are passing in the CI. Here the logs in the CI:
```
go test ./tracer/...
ok  	github.com/DataDog/dd-trace-go/tracer	0.081s <-- that's running
ok  	github.com/DataDog/dd-trace-go/tracer/contrib/gin-gonic/gintrace	0.016s
ok  	github.com/DataDog/dd-trace-go/tracer/contrib/gorilla/muxtrace	0.007s
ok  	github.com/DataDog/dd-trace-go/tracer/contrib/tracegrpc	0.028s
?   	github.com/DataDog/dd-trace-go/tracer/ext	[no test files]
```

and the Coverage:
```
github.com/DataDog/dd-trace-go/tracer/transport.go:101:    changeEncoder  100.0%
github.com/DataDog/dd-trace-go/tracer/transport.go:111:    apiDowngrade   100.0%
```

I'm sorry if it's not so friendly running integration tests at the moment, but we will address that soon releasing the agent open source or at least a docker container with the binary.

[1]: https://github.com/DataDog/dd-trace-go/blob/master/circle.yml#L21